text/0000-create-language-family-builders.md

```diff
@@ -0,0 +1,148 @@
+
+## Motivation
+
+The current builders have nearly reached the maximum layer limit for container images making it challenging to add new buildpacks. The ultimate goal is to include additional JVM providers in the Java composite buildpack as proposed in [#267 Add additional JVMs to the Java buildpack](https://github.com/paketo-buildpacks/rfcs/pull/267).  
+
+Additionally, a recently released version of the Full Builder (v0.2.278) contains 103 layers. When a user builds with the Full Builder building their app, all 103 layers must be pulled down locally.  The size of the builder and time to pull it down is considerable.  Using a language-specific builder we can greatly reduce the resources (size and speed) needed to run `pack build` and improve the user experience.  As a result, we encourage adoption of the Paketo buildpacks.
+
+Finally, there are a set of APM buildpacks that are not natively available in the current builders.  A language family builder can include the APM buildpacks to make it easier for users to utilize these tools and facilitate adoption.
+
+## Detailed Explanation
+The maximum number of layers an image can have is 127.  The current full builder contains 103 layers providing little room for additional layers.  
+A Java-specific builder with all 9 currently available JVM providers would contain 44 layers.  Each of the APM buildpacks would add an additional layer.
+
+## Rationale and Alternatives
+
+One alternative is compressing the layers in the current builders as proposed in the [upstream RFC](https://github.com/buildpacks/pack/issues/1595).  
+
+Another alternative is adding fewer additional JVM provideds to the existing builders.  Each buildpack adds one layer to the image.  If the 8 other JVM providers were included in the Java buildpack the number of layers would increase to 111.  Still within the 127 max but inching closer. Perhaps a smaller set of JVM providers could be added instead of all 8.
+
+A third alternative is to create a JVM meta-buildpack that contains all the JVM provider buildpacks and configures the JRE/JDK dependencies.   This would replace the current BellSoft Liberica buildpack.  
+
+However, for all 3 alternatives, a language-specific builder still reduces the size (number of layers in the builder) and speed (downloading fewer buildpacks) of building applications and provides growing room to add additional buildpacks like the APMs.    
+
+## Implementation
+
+This RFC proposes that the builder team own the language-specific builders.  The current process of updating the Paketo builders is automated using a builder.toml. Adding additional language-specific builder config files in each builder repo therefore has little impact on maintenance cost for the project.
```

Comment:
Is this OK with @paketo-buildpacks/builders-maintainers? Is it technically possible to add additional builders this way?

Also, I would kind of like to see this jointly owned. Where the @paketo-buildpacks/builders-maintainers own the machinery/workflows to produce the images, but the language team owns the actual builder.toml & any tests for validation.

Thoughts?

Reply:
I think we should also make a note here to say something like, basically to make it clear that ownership is delegated to the language family teams. RFCs and decisions about these builders would go through the language family RFC process.

```
The decision to create language family-specific builders is owned by each language family team. Teams may decide to create zero or more based on their needs and user demand. In addition, each language-family team owns the decisions about what goes into the builder. Again, this can be based on team and user needs.
```

Reply:
@dmikusa I have updated the RFC to more clearly propose joint ownership and delegate the creation to each individual language family.

Reply:
Thanks Kevin. I'm going to resolve this conversation. I've added @paketo-buildpacks/builders-maintainers as a reviewer, so we can get some feedback.